databroker/assets/handlers.py

```diff
@@ -203,6 +203,105 @@ def __init__(self, filename, frame_per_point=1):
             frame_per_point=frame_per_point)
 
 
+class AreaDetector_HDF5SingleHandler(HandlerBase):
+    '''Handler for hdf5 data stored 1 image per file and returned as a
+    numpy.array.
+
+    Parameters
+    ----------
+    fpath : string
+        filepath
+    template : string
+        filename template string.
+    filename : string
+        filename
+    key : string
+        the 'path' inside the file to the data set.
+    column_names : list[str]
+        The column names of the table
+    frame_per_point : float
+        the number of frames per point.
+    '''
+    specs = {'AD_HDF5_SINGLE'} | HandlerBase.specs
+
+    def __init__(self, fpath, template, filename, key='/entry/data/data',
+                 frame_per_point=1):
+        # I have included defaults for `key` for back compatibility.
+        self._path = os.path.join(fpath, '')
+        self._fpp = frame_per_point
+        self._template = template
+        self._filename = filename
+        self._key = key
+
+    def _fnames_for_point(self, point_number):
+        start = int(point_number * self._fpp)
+        stop = int((point_number + 1) * self._fpp)
+        for j in range(start, stop):
+            yield self._template % (self._path, self._filename, j)
+
+    def __call__(self, point_number):
+        ret = []
+        import h5py
+        for fn in self._fnames_for_point(point_number):
+            with h5py.File(fn, 'r') as f:
+                array = np.array(f[self._key][:])
+            ret.append(array)
+        return ret
+
+    def get_file_list(self, datum_kwargs):
+        ret = []
+        for d_kw in datum_kwargs:
+            ret.extend(self._fnames_for_point(**d_kw))
+        return ret
+
+
+class AreaDetector_HDF5SingleHandler_DataFrame(AreaDetector_HDF5SingleHandler):
+    '''Handler for hdf5 data stored 1 image per file and returned as a
+    Pandas.DataFrame.
+    This will work with all hdf5 files that are a mxn arrays and the data is
+    'table like' where m is the number of columns and n is the number of rows.
+    Parameters
+    ----------
+    fpath : string
+        filepath
+    template : string
+        filename template string.
+    filename : string
+        filename
+    key : string
+        the 'path' inside the file to the data set.
+    column_names : list[str]
+        The column names of the table
+    frame_per_point : float
+        the number of frames per point.
+    '''
+    specs = {'AD_HDF5_SINGLE_DATAFRAME'} | HandlerBase.specs
+
+    def __init__(self, fpath, template, filename, key='/entry/data/data',
+                 column_names=None, frame_per_point=1):
+        # I have included defaults for `key` and 'column_names' for back
+        # compatibility with existing files at SIX.
+        super(AreaDetector_HDF5SingleHandler_DataFrame, self).__init__(
```

Comment:
I don't understand how it works here, can you explain why `AreaDetector_HDF5SingleHandler_DataFrame` is used within itself, and then the `self` is added?

Reply:
I was following the structure [here](https://github.com/NSLS-II/databroker/blob/f4fbfd352d6ab6ffbd494e9e8d948764423abee4/databroker/assets/handlers.py#L201). After testing I found that it was not necessary to add that so it has been removed.